File: 백준/Bronze/17950. 스노우볼/스노우볼.java
import java.io.*;
import java.util.*;

public class Main{
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer st = new StringTokenizer(br.readLine().trim());
		
		long H = Long.parseLong(st.nextToken()), x = Long.parseLong(st.nextToken());
        long mod = 1000000007;

        long result = 0;
        long power = 1;

        for (long i = 0; i < H; i++){
            long c = Long.parseLong(br.readLine().trim());
            power = (power * x) % mod;
            result = (result + c * power) % mod;
        }

        bw.write(result + "\n");

		bw.flush();
		bw.close();
		br.close();
	}
}

File: 백준/Bronze/17950. 스노우볼/스노우볼.py
import sys

input = sys.stdin.readline

H, x = map(int, input().strip().split())
result = 0
MOD = 10 ** 9 + 7

for i in range(H):
    c = int(input().strip()) 
    result += (c  * pow(x, i+1, MOD))

result %= MOD

sys.stdout.write(f"{result}\n")